### PR TITLE
Use octokit rather than sending emails

### PR DESCRIPTION
**Pull request by dfaulken**

### CODE REVIEW

File: config/config.json.example [hunk 1]
```diff
@@ -1,7 +1,5 @@
 {
   "stops": ["Studio Arts Building", "Fine Arts Center", "Haigis Mall"],
-  "mail_settings": {"to": "transit-it@admin.umass.edu",
-                   "from": "transit-it@admin.umass.edu",
-                   "subject": "PVTA realtime feed error"},
+  "github_token": "9fe5e3055d8f4e2fd70180cf1fc1616db6b2b948",
```
Comment (dfaulken) [inline config/config.json.example:3]: This is not a real GH access token, just a random hex string from https://random.org.

File: lib/bojangles.rb [hunk 2]
```diff
@@ -83,18 +83,18 @@ def cache_error_messages!(current_errors)
...
   # Only the first two lines of an error message need to match for us to decide
   # that we've already sent it.
   def compare_errors(current_errors, old_errors)
     new_errors = current_errors.reject do |error|
-      old_errors.any? { |old| error.lines.first(2) == old.lines.first(2) }
+      old_errors.keys.any? { |old| error[:message].lines.first(2) == old.lines.first(2) }
     end
     resolved_errors = old_errors.reject do |error|
-      current_errors.any? { |new| error.lines.first(2) == new.lines.first(2) }
+      current_errors.any? { |new| error.lines.first(2) == new[:message].lines.first(2) }
```
Comment (werebus) [inline lib/bojangles.rb:97]: GitHub's _syntax-highlighting_ isn't a fan of using `new` as a variable name, but this fine, I'm pretty sure.

File: lib/bojangles.rb [hunk 3]
```diff
@@ -104,16 +104,22 @@ def go!
     errors = DepartureComparator.compare
     current_time = Time.now
     new_errors, resolved_errors = compare_errors(errors, cached_error_messages)
+    issue_numbers = cached_error_messages
```
Comment (werebus) [inline lib/bojangles.rb:107]: So, this variable contains more than just the "numbers", right?

File: lib/bojangles.rb [hunk 4]
```diff
@@ -123,10 +129,7 @@ def go!
 
   # rubocop:disable Style/IfUnlessModifier
   def message_html(new_errors, resolved_errors)
-    message = [<<~MESSAGE.tr("\n", ' ')]
-      This message brought to you by Bojangles, UMass Transit's monitoring
-      service for the PVTA realtime bus departures feed.
-    MESSAGE
+    message = ''
```
Comment (werebus) [inline lib/bojangles.rb]: Is this method actually used anywhere?

File: lib/departure_comparator.rb [hunk 5]
```diff
@@ -74,32 +74,41 @@ def email_format(time)
...
     if other_headsigns.length == 1
-      message += "  Found alternative: #{other_headsigns.first}"
+      message += "Found alternative: #{other_headsigns.first}"
     elsif other_headsigns.length > 1
-      message += "  Found alternatives: #{other_headsigns.join(', ')}"
+      message += "Found alternatives: #{other_headsigns.join(', ')}"
     end
-    @messages << message
+    issue[:message] = message
+    @messages << issue
```
Comment (werebus) [inline lib/departure_comparator.rb]: Similar naming nit-pick. This instance variable, `@messages` contains `issue`s, which each have a `[:message]`

**Review by werebus — CHANGES_REQUESTED** [review]
Man, GitHub's syntax highlighting hates HEREDOCs.

Mostly naming nit-picks...

File: lib/bojangles.rb (hunk 3 above)
Comment (dfaulken) [reply lib/bojangles.rb:107]: Yes, it's a mapping from message text to GH issue numbers.

File: lib/bojangles.rb (hunk 4 above)
Comment (dfaulken) [reply lib/bojangles.rb]: d880367

File: lib/departure_comparator.rb (hunk 5 above)
Comment (dfaulken) [reply lib/departure_comparator.rb]: 2f44523

File: lib/bojangles.rb (hunk 2 above)
Comment (dfaulken) [reply lib/bojangles.rb:97]: My understanding is that it's only interpreted as a keyword when followed by a class name.

**Comment by mboneil10** [comment]
Does this PR affect the tests?

**Comment by dfaulken** [comment]
Yes, I just remembered a few minutes ago that bojangles has tests. I'm working on them now. My thinking is to open a separate PR for that to avoid bulking up this one.

File: lib/bojangles.rb [hunk 6]
```diff
@@ -104,58 +104,29 @@ def go!
...
+      client = Octokit::Client.new access_token: GITHUB_TOKEN
+      new_errors.each do |error|
+        title, message = error.values_at :title, :message
+        message_text = [Time.now.strftime('%l:%M %P'), message].join ': '
+        issue = client.create_issue 'umts/realtime-issues', title, message_text
+        issue_numbers[message] = issue.number
+      end
+      resolved_errors.each_pair do |message, issue_number|
+        comment = "#{Time.now.strftime('%l:%M %P')}: This error is no longer present."
+        client.add_comment 'umts/realtime-issues', issue_number, comment
```
Comment (mboneil10) [inline lib/bojangles.rb:118]: That's awesome

**Review by mboneil10 — APPROVED** [review]
Yay, octokit to the rescue!

**Review by werebus — APPROVED** [review]

**Pull request merged**